### DISCUSSION TITLE
0607 피어세션일지

### DISCUSSION THREAD

**Discussion by gvsteve24** [bcaitech1/p4-dkt-freshtomato#82] [General]
# 주말을 이모지로 표현해봐용!

| 진현          | 아라         | 정현        | 민용        | 재훈           | 유라           |
| ------------- | ------------ | ----------- | ----------- | -------------- | -------------- |
| :green_salad: | :heart_eyes: | :cockroach: | :wow_amaze: | :man_mechanic: | :yawning_face: |

* **[TMI]** 아라의 남자친구분은 제주도에 사시는 분 ㅎㅅㅎ

# 회의 및 토론

* ### 현재 진행 상황

**[아라]** Last Query 피처추가하면서 돌려봄
[Minon feature만 사용하여 Last Query 실험]결과가 잘 안좋았다. SAINT로 돌려보려고 하고 있음
**[민용]** SAINT로 했을 때는 valid 평균 했을 때 0.812정도 나왔었음

**[재훈]** 
Riiid train data는 만들어놨고 돌리는건 아직 안해봄, 나머지는 머신러닝 쪽 돌리는 중! 
윈도우로 마지막 유저를 포함한 관계에 대해서 피처 추가중 
(shift로 concat해서 붙여봤는데 더 효율적으로 활용하는 방향 모색중)

**[민용]**

1. **[Augmentation](https://github.com/bcaitech1/p4-dkt-freshtomato/discussions/74)**

   Data augmentation을 진행했는데 valid auc는 0.84정도로 나오고 lb에서는 오히려 더 낮게 나왔음

   **data augmentation이 잘못 만들어진건지?**

   [아라] userID 가..train과 valid랑 같이 존재할 수 있는 상황아닐까?

   [민용] user와는 큰 관계가 없는 feature를 썼다고 해서 user가 섞이는 방향은 아닐거라고 생각하는데.. 보완하는 방법을 잘 모르겠다.

2. **추가 실험 상황**

+ 문제별로 몇초 걸리는지(elapsed time) 추가 => increased valid auc little bit

* optimizer adamw => increased valid auc 
* 인코더 멀티헤드 스택킹 다양하게 바꿔도 굿
  ( 제일 작은 모델이 현재까지는 제일 좋았음 )
* Saint encoder decoder들어가는거 같이 맞춰서 실험

* decoder에는 interaction만 넣어봄

3. **추가 추가**

* encoder decoder에 넣기전에 convolution layer를 써보려고 생각중
* 큰모델은 안써도 될거같긴하다
* GRU att 잘쓰고 잇는데 최대한 작게 작게 할수있는지 생각중

**[유라]** feature이름 변경해서 함수내에 잘 넣게 수정했고, Pycaret에 custom KFold 넣어둔 상태, 공유예정

**[진현]** 문제의 난이도를 정하는 방법 ([문항반응이론](https://ko.wikipedia.org/wiki/%EB%AC%B8%ED%95%AD_%EB%B0%98%EC%9D%91_%EC%9D%B4%EB%A1%A0))
SAINT를 베이스라인을 수정하고 있음

**[정현]** LSTM ATT 를 더 보기로 함 ! 다른 조에서 잘 나오는 거 같음

**[유라 추가]**
K-fold로 유저당 마지막 row를 validation으로 쓰는거를 돌리고 있음, 돌아감!
재훈_피처에 nan값 안채운걸 validation 기준바꾼걸로 제출해보겠습니다.
validation이랑 fold-out 이랑 차이가 많이 남. 0.81 vs 0.83 차이

**[정형 컴피티션]** validation은 계속 오르지만 LB는 그대로였던 패턴이 있는데 지금은 test분포가 우리한테 주어진 상황 && kfold 가 올라가면 같이 올라가는 상황이라 아직은 단정하기 이름

**[민용]** seed 1997&42 안정적으로 LB, Valid 비슷하게 나옴

### 다른 조 참고!

* dktㅈㅎㅈ : LSTM ATTention + 피처 추가! 로 잘하는가봄

* max_seq_length를 데이터마다 랜덤으로 부여했을때 전체 시퀀스를 기준으로 괜찮음!
  소요시간을 범주형 / 연속형 둘다 부여해 씀
  convbert 를 쓰고있나봄! 

**고민**

**[정현]** 베이스라인을 반복해서 보더라도 자세히 몰라 디버깅이 안됩니다..

**[아라]** 디스커션에 코드를 공유해서 조언을 받아보세요 :smile:

**[재훈]** Riiid pretrain할 데이터랑 코드 올릴 예정

**[제안]** 내일 이전까지는 모델관련한 강의 꼭 보고 이야기 나눌 수 있도록 합시다

# 저녁메뉴

| 진현          | 아라       | 정현         | 민용       | 재훈       | 유라       |
| ------------- | ---------- | ------------ | ---------- | ---------- | ---------- |
| :green_salad: | :green_salad: | 간장제육덮밥 | 보쌈정식(츄릅) | :question: | :question: |

**Comment by arabae** [comment]
**TODO**

- [x] Last Query 수정된 feature를 이용하여 실험
   - [x] 이전에 수행했던 LSTM, ~LSTMATTN~에도 수정된 feature를 이용하여 재실험
- [x] continuous feature를 하나씩 추가하면서 AUC 확인
- [ ] Last Query 전체 동작 과정 스케치하고, 아이디어 생각하기
- [ ] **serving 수업 듣고, 따라하기** ⭐

**Comment by gvsteve24** [comment]
TODO

- [x] Cuda error 수정
- [ ] LSTM-Att 구조, 인풋 파악하기
- [ ] LSTM-Att 실험하기
- [ ] [정리] 위 내용 짧게 정리하기

**Comment by yukyunglee** [comment]
다들 아주 잘하고 계시는 군요 +_+
